src/main.ts

```diff
@@ -169,3 +174,67 @@ app.on('activate', () => {
     createWindow()
   }
 })
+
+ipcMain.on('app-ready', () => {
+  if (app.isPackaged && checkSignatureExists()) {
+    const feedURL = `${pkg.hostUrl}/${pkg.repoName}/${process.platform}-${
+      process.arch
+    }/${app.getVersion()}`
+
+    autoUpdater.setFeedURL({
+      url: feedURL,
+      serverType: 'default',
+    })
+    autoUpdater.checkForUpdates()
+    setInterval(() => {
+      autoUpdater.checkForUpdates()
+    }, 4 * 60 * 60 * 1000)
+  }
+})
+
+ipcMain.on('restart_app', () => {
+  autoUpdater.quitAndInstall()
+})
+
+autoUpdater.on('checking-for-update', () => {
+  console.log('checking-for-update')
+})
+
+autoUpdater.on('update-available', () => {
+  console.log('update-available')
+})
+
+autoUpdater.on(
+  'update-downloaded',
+  (event, releaseNotes, releaseName, updateURL) => {
+    if (mainWindow && mainWindow.webContents) {
+      mainWindow.webContents.send('update_downloaded')
+    }
+    console.log('update-downloaded', {
+      event,
+      releaseNotes,
+      releaseName,
+      updateURL,
+    })
+  },
+)
+
+autoUpdater.on('update-not-available', () => {
+  console.log('update-not-available')
+})
+
+autoUpdater.on('error', err => {
+  console.log('error', { err })
+})
+
+const checkSignatureExists = () => {
+  if (os.platform() === 'darwin') {
+    const certExists = fs.existsSync(
+      path.join(__dirname, 'static', 'entitlements.plist'),
+    )
```

Comment:
This is not a certificate. This is an MacOS app permission manifest. Not sure if it's valid to use it for detecting signature

Reply:
Yeah, I think we don't need it. Removed